auth/services/selfsigned/web/handler.go

```diff
@@ -0,0 +1,128 @@
+package web
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/labstack/echo/v4"
+	"github.com/nuts-foundation/nuts-node/auth/contract"
+	"github.com/nuts-foundation/nuts-node/auth/log"
+	"github.com/nuts-foundation/nuts-node/auth/services/selfsigned/types"
+	"github.com/nuts-foundation/nuts-node/core"
+	"html/template"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+)
+
+type Handler struct {
+	store types.SessionStore
+}
+
+func NewHandler(store types.SessionStore) *Handler {
+	return &Handler{store}
+}
+
+// Routes registers the Echo routes for the API.
+func (h Handler) Routes(router core.EchoRouter) {
+	RegisterHandlers(router, h)
+}
+
+func (h Handler) RenderEmployeeIDPage(ctx echo.Context, sessionID string, params RenderEmployeeIDPageParams) error {
+	session, ok := h.store.Load(sessionID)
+	if !ok {
+		return echo.NewHTTPError(http.StatusNotFound, "session not found")
+	}
+
+	// find the correct template language from the used contract template
+	userContract, err := contract.ParseContractString(session.Contract, contract.StandardContractTemplates)
+	if err != nil {
+		return err
+	}
+	lang := userContract.Template.Language
+
+	wd, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+	tmpl, err := template.ParseFiles(fmt.Sprintf("%s/auth/services/selfsigned/web/templates/employee_identity_%s.html", wd, strings.ToLower(string(lang))))
+	if err != nil {
+		return err
+	}
```

Comment:
weird stuff, why not just embed it? And if you want to allow it to be customized, just let the admin configure a directory. Working directory is highly dependent on how the application is started.

Reply:
fixed